src/backend/main.ts

```diff
@@ -1168,6 +1176,75 @@ ipcMain.handle(
   }
 )
 
+async function runBeforeLaunchScript(
+  gameInfo: GameInfo,
+  gameSettings: GameSettings
+) {
+  if (gameSettings.beforeLaunchScriptPath) {
+    return new Promise((resolve, reject) => {
+      logInfo(
+        [
+          'Running script before',
+          gameInfo.title,
+          `(${gameSettings.beforeLaunchScriptPath})`
+        ],
+        LogPrefix.Backend
+      )
+
+      // Execute script before launch, wait until the script
+      // exits to continue
+      //
+      // The script can start sub-processes with `bash another-command &`
+      // if `another-command` can run async
+      const child = spawn(gameSettings.beforeLaunchScriptPath, {
+        cwd: gameInfo.install.install_path,
+        ...execOptions
+      })
+
+      child.stdout.on('data', (data) => {
+        logInfo(data.toString(), LogPrefix.Backend)
+      })
+
+      child.stderr.on('data', (data) => {
+        logInfo(data.toString(), LogPrefix.Backend)
+      })
+
+      child.on('exit', () => {
+        resolve(true)
+      })
+
+      child.on('error', (err: Error) => {
+        logError(
+          ['Error running before script: ', err.message],
+          LogPrefix.Backend
+        )
+        if (err.stack) logError(err.stack, LogPrefix.Backend)
```

Comment:
You can just pass the `err` directly to `logError`, it already outputs either its `stack` or `message` depending on which is present (and the `stack` includes the `message`)